NuWeather/plugin.py

```diff
@@ -459,17 +459,37 @@ def _darksky_fetcher(self, location):
                           'summary': forecastdata['summary'].rstrip('.')} for idx, forecastdata in enumerate(data['daily']['data'])]
         }
 
-    @wrap([getopts({'user': 'nick', 'backend': None, 'forecast': ''}), additional('text')])
+    def _get_valid_backends(self):
+        """Fetches valid backends by checking to see if API keys are set in registry"""
+        w_backends = [backend for backend in BACKENDS if self.registryValue('apikeys.{}'.format(backend))]
+        g_backends = [backend for backend in GEOCODE_BACKENDS if self.registryValue('apikeys.{}'.format(backend))]
+
+        return {'weather': w_backends, 'geo': g_backends}
+
+
+    @wrap([getopts({'user': 'nick', 'weatherbackend': None, 'geobackend': None, 'forecast': '', 'list': ''}), additional('text')])
     def weather(self, irc, msg, args, optlist, location):
-        """[--user <othernick>] [--backend <backend>] [--forecast] [<location>]
+        """[--user <othernick>] [--weatherbackend <weather backend>] [--geobackend <geocode backend>] [--forecast] [<location>]
 
         Fetches weather and forecast information for <location>. <location> can be left blank if you have a previously set location (via 'setweather').
 
         If --forecast is specified, show an extended (default: 5-day) forecast for <location>.
 
         If the --user option is specified, show weather for the saved location of that nick, instead of the caller's.
+
+        If either --weatherbackend or --geobackend is specified, will override the default backends if provided backend is available.
+
+        If --list is specified, will list all available backends.
         """
         optlist = dict(optlist)
+        valid_backends = self._get_valid_backends()
+
+        # Check for --list option first
+        if optlist.get('list'):
+            irc.reply("Weather Backends: {0}".format(", ".join(valid_backends['weather'])))
+            irc.reply("Geocode Backends: {0}".format(", ".join(valid_backends['geo'])))
+            return
```

Comment:
Using multiple irc.reply is generally discouraged since it overrides `oneToOne` Supybot options; in this case the data is short and can just fit on one line.